maestro/audit.py

```diff
@@ -351,6 +358,57 @@ def from_config(cfg):
             cfg.get('timeout', WebHookAuditor.DEFAULT_TIMEOUT))
 
 
+class ExecuteScriptAuditor(BaseAuditor):
+    """Auditor that executes scripts with maestro action information."""
+
+    def __init__(self, script, args, level):
+        super(ExecuteScriptAuditor, self).__init__(level)
+        if not script:
+            raise exceptions.InvalidAuditorConfigurationException(
+                'Missing script to execute!')
+
+        self._script = script
+        self._args = args
+
+    def _format_container_dict(self, what):
+        if isinstance(what, entities.Entity):
+            return None
+        d = {}
+        for item in what:
+            x = dict(ship=item.ship.ip, service=item.service.name)
+            d[item.name] = x
+        return StringIO(json.dumps(d)).read()
```

Comment:
If you don't need a file-like object (because you don't pass it via the `stdin` argument to `Popen`) then you don't need to use `StringIO` at all. Just `return json.dumps(d)`.